### PR TITLE
Pgpool backend reference change

### PR DESCRIPTION
**Pull request by MobarakHsn**
Signed-off-by: MobarakHsn <mobarak@appscode.com>

### CODE REVIEW

File: pgpool/kubedb_client_builder.go [hunk 1]
```diff
@@ -106,23 +106,23 @@ func (o *KubeDBClientBuilder) getURL() string {
...
-			return "", "", fmt.Errorf("backend database auth secret not found")
-		}
-		secretName = apb.Spec.Secret.Name
+	if pp.Spec.PostgresRef == nil {
+		return "", "", fmt.Errorf("there is no postgresRef found for pgpool %s/%s", pp.Namespace, pp.Name)
+	}
+	apb := &appbinding.AppBinding{}
+	err := o.kc.Get(o.ctx, types.NamespacedName{
+		Name:      pp.Spec.PostgresRef.Name,
+		Namespace: pp.Spec.PostgresRef.Namespace,
```
Comment (TasdidurRahman) [inline pgpool/kubedb_client_builder.go:115]: cross namespace grant should be checked

Comment (TasdidurRahman) [reply pgpool/kubedb_client_builder.go:115]: ignore for this release

File: pgpool/kubedb_client_builder.go [hunk 2]
```diff
@@ -106,23 +106,23 @@ func (o *KubeDBClientBuilder) getURL() string {
...
+	}, apb)
+	if err != nil {
+		return "", "", err
 	}
+	if apb.Spec.Secret == nil {
+		return "", "", fmt.Errorf("backend postgres database auth secret not found for pgpool %s/%s", pp.Namespace, pp.Name)
```
Comment (TasdidurRahman) [inline pgpool/kubedb_client_builder.go:121]: "backend auth secret unspecified"

**Review by TasdidurRahman — CHANGES_REQUESTED** [review]
update error msg

**Review by TasdidurRahman — APPROVED** [review]

**Review by raihankhan — APPROVED** [review]
LGTM!

**Pull request merged**